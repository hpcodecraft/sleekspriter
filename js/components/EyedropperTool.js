var EyedropperTool = React.createClass({
  render: function() {
    return (
      <div id="Eyedropper-Tool" className="ToolComponent">
        <i className="icon flaticon-eyedropper2"></i>
        <div id="EyedropperSwatch" className="colorswatch" style={{background: this.props.editor.pixelColor.rgbaString()}}></div>
        <ul>
          <li>Hex: {this.props.editor.pixelColor.alpha() == 0 ? '': this.props.editor.pixelColor.hexString()}</li>
          <li>RGB: {this.props.editor.pixelColor.alpha() == 0 ? '': this.props.editor.pixelColor.red()+', '+this.props.editor.pixelColor.green()+', '+this.props.editor.pixelColor.blue()}</li>
        </ul>
        <span className="spacer"></span>
        <span className="hint">Click any non-transparent pixel to pick its color.</span>

      </div>
    );
  }
});