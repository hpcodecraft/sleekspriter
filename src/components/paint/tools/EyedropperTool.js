import React from "react";
import { t } from "../../../utils";

const EyedropperTool = (props) => {
  return (
    <div id="Eyedropper-Tool" className="ToolComponent">
      <i className="icon flaticon-eyedropper2"></i>
      <div id="EyedropperSwatch" className="colorswatch"></div>
      <ul>
        <li>{t("Hex")}: <span id="EyedropperHex">{t("transparent")}</span></li>
        <li>{t("RGB")}: <span id="EyedropperRGB">-, -, -</span></li>
      </ul>
      <span className="spacer"></span>
      <span className="hint">{t("Click any non-transparent pixel to pick its color.")}</span>
    </div>
  );
};

export default EyedropperTool;
