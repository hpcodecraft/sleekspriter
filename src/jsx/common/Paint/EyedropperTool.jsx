/** @jsx React.DOM */
var EyedropperTool = React.createClass({
  render: function() {
    var color = this.props.editor.color.frame;
    return (
      <div id="Eyedropper-Tool" className="ToolComponent">
        <i className="icon flaticon-eyedropper2"></i>
        <div id="EyedropperSwatch" className="colorswatch" style={{background: color.rgbaString()}}></div>
        <ul>
          <li>Hex: {color.alpha() == 0 ? '': color.hexString()}</li>
          <li>RGB: {color.alpha() == 0 ? '': color.red()+', '+color.green()+', '+color.blue()}</li>
        </ul>
        <span className="spacer"></span>
        <span className="hint">Click any non-transparent pixel to pick its color.</span>

      </div>
    );
  }
});