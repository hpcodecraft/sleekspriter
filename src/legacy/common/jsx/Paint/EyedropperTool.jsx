var EyedropperTool = React.createClass({
  render: function() {
    return (
      <div id="Eyedropper-Tool" className="ToolComponent">
        <i className="icon flaticon-eyedropper2"></i>
        <div id="EyedropperSwatch" className="colorswatch"></div>
        <ul>
          <li>Hex: <span id="EyedropperHex">transparent</span></li>
          <li>RGB: <span id="EyedropperRGB">-, -, -</span></li>
        </ul>
        <span className="spacer"></span>
        <span className="hint">Click any non-transparent pixel to pick its color.</span>
      </div>
    );
  }
});
