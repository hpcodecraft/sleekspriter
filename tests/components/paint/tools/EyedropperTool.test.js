import React from "react";
import EyedropperTool from "components/paint/tools/EyedropperTool";

describe("EyedropperTool", () => {
  it("should render correctly", () => {
    const component = shallow(<EyedropperTool />);
    expect(component.html()).toMatchSnapshot();
  });
});
