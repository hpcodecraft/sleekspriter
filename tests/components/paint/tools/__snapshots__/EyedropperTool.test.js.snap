// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EyedropperTool should render correctly 1`] = `
<div
  class="ToolComponent"
  id="Eyedropper-Tool"
>
  <i
    class="icon flaticon-eyedropper2"
  />
  <div
    class="colorswatch"
    id="EyedropperSwatch"
  />
  <ul>
    <li>
      Hex: 
      <span
        id="EyedropperHex"
      >
        transparent
      </span>
    </li>
    <li>
      RGB: 
      <span
        id="EyedropperRGB"
      >
        -, -, -
      </span>
    </li>
  </ul>
  <span
    class="spacer"
  />
  <span
    class="hint"
  >
    Click any non-transparent pixel to pick its color.
  </span>
</div>
`;
